automation/gen_proto_py.py

```
@@ -0,0 +1,109 @@
+
+def run_protoc(proto_files: list[Path]) -> None:
+    """Run protoc to generate Python code."""
+    cmd = [
+        sys.executable,
+        "-m",
+        "grpc_tools.protoc",
+        f"-I{PROTO_DIR}",
+        f"--python_out={OUTPUT_DIR}",
+    ] + [str(f) for f in proto_files]
+
+    result = subprocess.run(cmd, capture_output=True, text=True)
```

Ruff (0.15.6)

[error] `subprocess` call: check for execution of untrusted input

(S603)